llvm/lib/SYCLLowerIR/SpecConstants.cpp

```diff
@@ -1029,6 +1026,9 @@ PreservedAnalyses SpecConstantsPass::run(Module &M,
   for (const auto &P : DefaultsMetadata)
     MDDefaults->addOperand(P);
 
+  if (Mode == HandlingMode::default_values)
```

Comment:
Conceptually, `HandlingMode` is both set outside of the pass and read outside of the pass (that property to indicate that a device image was produced with default values of spec constants), so it seems to me that that metadata is better to be set somewhere outside the pass.

Reply:
yeah i debated this, let me try putting it somewhere when we do the default split, will do in follow-up pr thx